tools/cpplint.py

```diff
@@ -1895,6 +1895,25 @@ def CheckForBadCharacters(filename, lines, error):
       error(filename, linenum, 'readability/nul', 5, 'Line contains NUL byte.')
 
 
+def CheckInlineHeader(filename, include_state, error):
+  """Logs an error if both a header and its inline variant are included."""
+
+  # Iterate over all headers, looking for any inline headers
+  for section_list in include_state.include_list:
+    for header in section_list:
+      # Proceed further for only inline headers
+      m = Match(r'^(.+)-inl.h$', header[0])
+      if m:
+        name = '%s.h' % m.group(1)
+        # Look for the corresponding header
+        for sl in include_state.include_list:
+          for h in section_list:
+            if h[0] == name:
+              err =  '%s includes both %s and %s' % (filename, header[0], h[0])
+              error(filename, h[1], 'build/include', 5, err)
+              print err
```

Comment:
More succinct of way of doing it:
```python
all_headers = dict(item for sublist in include_state.include_list
                   for item in sublist)
bad_headers = set('%s.h' % name[:-6] for name in all_headers.keys()
                  if name.endswith('-inl.h'))
bad_headers &= set(all_headers.keys())  # compute intersection

for name in bad_headers:
  # lineno = all_headers[name]
```
It's also slightly more idiomatic vis-a-vis other include checks.

Reply:
@bnoordhuis okay, will try this. That said, I'm more interested at this point to find out what's wrong with my solution? It might not be idiomatic, but it should work. Why does `error` not work?

Reply:
Oh, is that your question? :-)

I assume you're testing it from the command line like this: `python tools/cpplint.py src/*.cc`?

Look for `_DEFAULT_FILTERS`: `build/include` is off by default.  Pass `--filter=build/include` to test it, or `--filter=+build/include` to enable it in addition to the other rules.

Reply:
@bnoordhuis I was! That helped.